lti_consumer/plugin/compat.py

```diff
@@ -121,3 +121,40 @@ def user_course_access(*args, **kwargs):
     # pylint: disable=import-error,import-outside-toplevel
     from lms.djangoapps.courseware.courses import check_course_access
     return check_course_access(*args, **kwargs)
+
+
+def get_or_create_externalid(user):
+    """
+    Given a list of user, returns corresponding external id's
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from openedx.core.djangoapps.external_user_ids.models import ExternalId
+    return ExternalId.add_new_user_id(user, 'lti')
+
+
+def get_course_members(*args, **kwargs):
```

Comment:
@shimulch I think we're messing too much with internals in the core in a way that is not providing a nice reusable function. I'm considering a few alternatives:
1. Keep the current implementation in the core.
2. Move the course members queryset function inside the `lti_consumer` repo and simplify it.
What do you think is the best approach?

Reply:
@giovannicimolin, the core API now follows the edX guideline to return serialized data.